apps/entity-service/src/entities/entities.service.ts

```diff
@@ -96,7 +119,10 @@ export class EntitiesService {
     await this.policyService.authorize(action, subject);
   }
 
-  async isFrameworkAdmin<T extends EntityModel>({ frameworkKey }: T) {
+  async isFrameworkAdmin<T extends EntityModel>(T: EntityModel) {
+    if (T instanceof Disturbance) return;
+    if (T instanceof Invasive) return;
+    const { frameworkKey } = T;
```

Comment:
A few problems here:
* You've changed the type of the model accepted by this function - it is meant to accept a value of type T, not base `EntityModel`. The param to the method should be something more like `(entityModel: T)`
* Disturbance and Invasive are not entity models, so I'm not really sure what's going on here.
* You've chagned the return type of the method from `boolean` to `boolean | undefined`. If this change is really necessary, the two early returns should explicitly return a boolean.